### PR TITLE
Abstract hash <-> pyobject conversion

### PR DESCRIPTION
**Pull request by betatim**
Add a layer of abstraction for converting hash values into python
objects (and back). This is in preparation for supporting our
own hash type that is not a plain integer.

---

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Is it documented in the `ChangeLog`?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Is the Copyright year up to date?

### CODE REVIEW

File: khmer/_khmer.cc [hunk 1]
```diff
@@ -109,6 +109,32 @@ extern "C" {
 
 /***********************************************************************/
 
+// Convert a hash to a python long object.
+static bool convert_HashIntoType_to_PyObject(const HashIntoType &hashval,
+                                             PyObject **value) {
```
Comment (betatim) [inline khmer/_khmer.cc]: Dilemma: would like to keep the interface similar to the other `convert_*` functions but doing this `PyObject **value` and `convert_HashIntoType_to_PyObject(h, &ptr_to_pyobj)` double take seems extremely ugly. Any better ideas?

**Comment by betatim** [comment]
Ready for review! @luizirber, @camillescott, @standage , or @ctb

File: khmer/_khmer.cc (hunk 1 above)
Comment (standage) [inline khmer/_khmer.cc]: I actually think it's fine.

**Comment by standage** [comment]
LGTM, but I'd feel more comfortable if someone with more experience with the CPython code gave the official sign off.

**Comment by ctb** [comment]
Looks good - update & merge!

**Pull request merged**